File: Core-backend-layer/Programming-fundamentals/Java/Threads/src/OrderProcessorRunnable.java
// Thread that implements the Runnable interface
class OrderProcessorRunnable implements Runnable {
    private final Product product;
    private final int orderQuantity;

    public OrderProcessorRunnable(Product product, int orderQuantity) {
        this.product = product;
        this.orderQuantity = orderQuantity;
    }

    @Override
    public void run() {
        // Simulate a process of processing an order
        product.decreaseQuantity(orderQuantity);
    }
}

File: Core-backend-layer/Programming-fundamentals/Python/Threads/src/order_processor_runnable.py
# In Python, there's no explicit Runnable interface as in Java.
# In Python, any class with a run method can be considered runnable.
class OrderProcessorRunnable:
    def __init__(self, product, order_quantity):
        self.product = product
        self.order_quantity = order_quantity

    def run(self):
        # Simulate a process of processing an order
        self.product.decrease_quantity(self.order_quantity)